bugs/remove-ddtrace-dependency

noxasch opened the pull request.

khun84 commented on lib/sidekiq_tracer/sidekiq/middleware/client/tracer.rb:
```diff
@@ -5,18 +5,8 @@ module Client
         class Tracer
           def call(worker_class, msg, queue, _redis_pool)
             msg['request_id'] ||= Current.web_request_id if defined?(Current) && Current.web_request_id.present?
-            dist_trace_id = SidekiqTracer.skip_distributed_trace_on&.call(worker_class, msg, queue) ? nil : msg['trace_id']
```
@noxasch 

1. leave a hook in the config for ppl to do this, sth like `custom_client_distributed_tracer` and invoke it here.
2. add this as an example in the readme
3. bump the minor version of the gem

khun84 commented on lib/sidekiq_tracer/sidekiq/middleware/client/tracer.rb:
```diff
@@ -5,17 +5,14 @@ module Client
-          def dd_trace(trace_id: nil)
-            corr = Datadog.tracer&.active_correlation
-            { 'trace_id' => trace_id.presence || corr&.trace_id, 'span_id' => corr&.span_id }
+          def custom_client_distributed_tracer(_worker_class, msg, queue)
```
@noxasch dont underscore the `_worker_class`

khun84 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,8 @@
+## Release v0.2.0
+
+1. Remove ddtrace direct dependency 
+2. Add `custom_client_distributed_tracer` hook
```
@noxasch bump the gem version!

noxasch committed:
Address the following changes
- remove ddtrace dependency
- implement custom_client_distributed_tracer hook
- bump minor version
